lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -308,6 +308,76 @@ static void createAsyncCopy(scf::ForOp &forOp, tt::LoadOp loadOp, Value alloc,
   loadOp.erase();
 }
 
+static void createTMAAsyncCopy(
+    scf::ForOp &forOp, tt::ExperimentalDescriptorLoadOp loadOp, Value alloc,
+    Value insertIdx, Value extractIdx, Value barrier, Value phase,
+    CoarseSchedule &schedule, CoarseSchedule::Cluster prefetchCluster,
+    llvm::MapVector<Operation *, LoadInfo> &loadToInfo, int numStages) {
+  assert(phase && "Phase value is required for TMA async copy.");
+  OpBuilder builder(forOp);
+  Value zero = builder.create<arith::ConstantIntOp>(forOp.getLoc(), 0, 32);
+  builder.setInsertionPoint(loadOp);
+  // TODO merge subview part with async cp.
```

Comment:
+1, but this can wait I guess

Reply:
ah that's bad copy paste. To be honest I'm not sure we want to do that as I it doesn't improve anything and makes the op more complicated :)